src/input_pipeline.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 Google LLC
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     https://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+
+import tensorflow as tf
+
+import input_utils
+
+
+def create_pretrain_dataset(input_patterns,
+                            input_config,
+                            model_config,
+                            batch_size,
+                            tokenizer,
+                            is_training=True,
+                            input_pipeline_context=None,
+                            output_fake_labels=True):
+  """Creates input dataset from (tf)records files for pretraining."""
+
+  name_to_features = {'image_data': tf.io.FixedLenFeature([], tf.string)}
```

Comment:
Nit: do we want to specify the shape here instead of [] (if they are resized to the same size)?

Reply:
Or are these the raw image data?

Reply:
They are raw image data, so the shapes are not always the same.

Reply:
I see. Thanks for the clarification.